Documentation updates

desrosj opened the pull request:
This splits out the documentation updates from #212 that were unrelated to the overall goals of that pull request.

desrosj committed:
Include double quotes for environment variables

Co-Authored-By: Javier Casares <javier@casares.org>

desrosj committed:
Guard against word splitting due to spaces

desrosj committed:
Extract unrelated documentation changes from #212

Co-Authored-By: Javier Casares <javier@casares.org>

desrosj committed:
Merge remote-tracking branch 'origin/master' into documentation-changes

desrosj committed:
Remove double quoting of variables.

This will be hanlded in #254 instead.

desrosj committed:
Copy over `.env.default` comment changes

Co-Authored-By: Javier Casares <javier@casares.org>

desrosj committed:
Include handbook link for requirements

Co-Authored-By: Javier Casares <javier@casares.org>

kittenkamala reviewed (changes requested):
Looks great! Thank you for doing all this, the clarifying comments etc.

This is the tiniest thing -- Line 154 of readme.md is still a bit awkward. Maybe it could say  "This example **uses the** `/home/wptestrunner/` folder..."

On line 156, 166 and 436 are we removing the `bash` part of ` ```bash ` because those commands are linux cli and not bash? (for example bash doesn't support a vim or vi command) -- or is there another reason? Should we put a ` ```linux cli ` highlight there instead?

Thanks!

desrosj committed:
Few more documentation fixes.

This makes the following improvements:
- Ensure each function, and file have a proper short description.
- Use third-person singular verbs for function summaries.
- Enforces the 80 character limit (120 when indentation is present) for DocBlocks.
- Code should be self-documenting. Avoid explaining every step of the code.
- Multi-line inline comments should start with `/*` not `/**`.

Co-Authored-By: Javier Casares <javier@casares.org>

desrosj committed:
Merge branch 'master' into documentation-changes

kittenkamala commented:
I see! I vote we leave them in in that case. Should we make a different PR for the readme? 

Re:

> Thanks! I still have a few more changes I'd like to make.
> 
> When I created this PR, I started by copying every documentation change unrelated to adding support for mult-php/multi-commit from the other pull request and my plan was to make any further changes for consistency, and pose some discussion points within the code here.
> 
> The README file is still on my to do list.
> 
> The reason for removing the syntax highlighting for `bash` there is not clear. The [commit that introduced this change](https://github.com/WordPress/phpunit-test-runner/pull/212/commits/592e6c7dab5d63e77989269d72738a0814056627) did not really explain. I am leaning towards reverting the removal of those unless a good reason presents itself. There possibly are a code examples where the language could be improved.

desrosj committed:
Merge branch 'master' into documentation-changes

kittenkamala reviewed (commented):
I know this is still in progress, but just noticed these:
